### ISSUE TITLE
[question] Conan 2.X and CI

### ISSUE THREAD

**Issue by upsilona** [conan-io/conan#13274]
### What is your question?

Hello everyone,

I'm in the process of upgrading my projects in order to use conan V2,

I was heavily using conan CPT in my CI, and I'm now updating my workflow as CPT is deprecated / not part of V2 (as far as I understand),

In CPT I was using the `CONAN_UPLOAD_DEPENDENCIES` environement variable, in order to push to my personal repository  (jfrog) all dependencies that were built during the build, extremly usefull to save time for later builds,

How could I reproduce this behavious with conan ?

My first idea was to analyse the log output in order to identify the dependencies that were build, however this sounds complicated (at least for me), and I'm not sure about how to proceed. Maybe using the `--format=json` of the `conan create` comamnd ? Unless there is an other / easier way ?

Thanks!

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @upsilona 

Thanks for your question. It is indeed relevant, and we are looking for feedback of use cases of CPT in order to provide/implement/document alternatives.

I have some questions and observations about your setup:
- Do you do a ``conan create --build=missing`` that might be missing more than 1 package?
- Definitely do not parse the text output, please read: https://docs.conan.io/2/introduction.html#stable
- The ``json`` output is indeed preferred over the text, and it will be more stable (in the future, the json output is still experimental in 2.0, as it will need a bit more of user feedback to stabilize it)
- Are you building in a clean cache in CI? If yes, have you considered just uploading everything that is in the cache?

**Comment by upsilona** [comment]
Hello @memsharded, and thank you for your reply,

I'm indeed using using `conan create --build=missing`, depending on the setup the missing dependencies can be 0 or more than 30,

Something I appreciated with CPT is the ability to only push the packages that were created during the build, as I'm heavily using cross-compilation, with several settings and options, I cannot simply rely on the reference (name / version / user / channel) but I need the revisions of the dependencies to only push the packages that were really created and not the packages that were already installed,

I'm facing a similar issue to push the package I'm building, I don't know (yet) how to only push the revision that were build, using `conan create`,

Regarding the cache, I'm using both clean cache setup (conan not even installed when the build is launched), and CI with a cache being reused, so that I cannot only push all the cache,

**Comment by memsharded** [comment]
> I'm facing a similar issue to push the package I'm building, I don't know (yet) how to only push the revision that were build, using conan create,

It is not really necessary. The default ``conan upload`` patterns will take care to upload only the latest created revision, and not the older ones. Not specifying it, or making it explicit with ``conan upload pkg/version#latest`` (literal "latest") will work to upload only the latest revision.

Let's add this ticket as a possible feature discussion for 2.1, maybe it is a custom command initially, but lets consider it.

**Comment by memsharded** [comment]
good news. the ``package-list`` feature will be merged for 2.0.7 in https://github.com/conan-io/conan/pull/13928, that will implement this flow (and more) without needing CPT:

```bash
conan create app --format=json > graph.json
conan list --graph=graph.json --format=json > pkglist.json
conan upload --list=pkglist.json -r=default
```
